### PR TITLE
Simulator: run on chibios instead of freertos.

### PR DESCRIPTION
**Pull request by mlyle**
Another day, another sim commit.  Tried to make this as clean as possible.  It seems to work well on both Linux and OSX; I've been using this for my development of the ground api for the past few days.

### CODE REVIEW

**Comment by mlyle** [comment]
Ah, some of the earlier stuff needs fixing for other targets, I think.. looking...

**Comment by mlyle** [comment]
Looks good.

**Comment by peabody124** [comment]
I'm not sure if you want to include it in this patch, but I think we should drop the "revolution" target. This was essentially a development board and I don't think anyone is still using it. Since that is what we use for simulation, it would make sense to totally convert it to the simulation target, especially now simulators are unified. e.g. ```make fw_simulation```

File: flight/PiOS/Common/pios_semaphore.c [hunk 1]
```diff
@@ -397,6 +397,6 @@ bool PIOS_Semaphore_Give_FromISR(struct pios_semaphore *sema, bool *woken)
 
 	return result;
 }
-#endif /* !defined(SIM_POSIX) && !defined(SIM_OSX) */
+#endif /* defined(PIOS_INCLUDE_FREERTOS) && (!defined(SIM_POSIX) && !defined(SIM_OSX)) */
```
Comment (peabody124) [inline flight/PiOS/Common/pios_semaphore.c]: Can you remove all occurrences of ```SIM_OSX```

File: flight/PiOS/Common/pios_semaphore.c [hunk 2]
```diff
@@ -341,7 +341,7 @@ bool PIOS_Semaphore_Give(struct pios_semaphore *sema)
 }
 
 /* Workaround for simulator version of FreeRTOS. */
-#if !defined(SIM_POSIX) && !defined(SIM_OSX)
+#if defined(PIOS_INCLUDE_FREERTOS) && (!defined(SIM_POSIX) && !defined(SIM_OSX))
```
Comment (peabody124) [inline flight/PiOS/Common/pios_semaphore.c]: This change is not needed. It is already in a FreeRTOS only conditional.

File: flight/PiOS/inc/pios_semaphore.h.save [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/**
```
Comment (peabody124) [inline flight/PiOS/inc/pios_semaphore.h.save]: this file was probably not intended

**Pull request merged**